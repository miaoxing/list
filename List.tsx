import * as React from "react";
import classNames from 'classnames';

export interface ListProps {
  as?: any;
  className?: string;

  /**
   * 缩进列表
   */
  indented?: boolean;

  /**
   * 紧凑列表
   */
  condensed?: boolean;

  /**
   * 无边框列表
   */
  borderless?: boolean;
}

const List = (props: ListProps) => {
  const {
    as: Component,
    className,
    indented,
    condensed,
    borderless,
    ...rest
  } = props;

  return (
    <Component
      {...rest}
      className={classNames(
        className,
        'list',
        indented && 'list-indented',
        condensed && 'list-condensed',
        borderless && 'list-border-less',
      )}
    />
  );
};

List.defaultProps = {
  as: 'ul',
};

List.Indented = (props: ListProps) => (<List indented={true} {...props}/>);

export default List;
export const ListIndented = List.Indented;
