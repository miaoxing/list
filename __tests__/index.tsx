import * as React from 'react';
import {render} from '@testing-library/react';
import List, {ListProps} from '../List';

describe('list', () => {
  test('List', () => {
    const result = render(<List/>);
    expect(result.container).toMatchSnapshot();
  });

  test('List as HTML Element', () => {
    const result = render(<List as="div"/>);
    expect(result.container).toMatchSnapshot();
  });

  test('List as Component', () => {
    const Component = (props: ListProps) => {
      return <span {...props}/>;
    };
    const result = render(<List as={Component}/>);
    expect(result.container).toMatchSnapshot();
  });
});
