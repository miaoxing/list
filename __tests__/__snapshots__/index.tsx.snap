// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list List 1`] = `
<div>
  <ul
    class="list"
  />
</div>
`;

exports[`list List as Component 1`] = `
<div>
  <span
    class="list"
  />
</div>
`;

exports[`list List as HTML Element 1`] = `
<div>
  <div
    class="list"
  />
</div>
`;
